File: src/Implementations/Saw.cpp
#include "Saw.h"
#include "GameState.h"

void Saw::UseItem(GameState& gameState) 
{ 
	gameState.gameStateManager.SetDamage(gameState.gameConfig.sawDamage);
	if (gameState.gameStateManager.GetStarter() == GameEnums::STARTER_HUMAN)
	{
		gameState.log.push_back("Uzyles pily (obrazenia razy 2)!");
	}
	else
	{
		gameState.log.push_back("Komputer uzyl pily (obrazenia razy 2)!");
	}
}

File: C#/src/Implementations/Saw.cs
public class Saw : Item
{
    public Saw() { }

    public void UseItem(GameState gameState)
    {
        gameState.gameStateManager.SetDamage(gameState.gameConfig.sawDamage);
        if (gameState.gameStateManager.GetStarter() == (int)GameEnums.Starter.STARTER_HUMAN)
        {
            gameState.log.Add("Uzyles pily (obrazenia razy 2)!");
        }
        else
        {
            gameState.log.Add("Komputer uzyl pily (obrazenia razy 2)!");
        }
    }
}